### PR TITLE
Added rationale section per #22886

### LINKED ISSUE #22886: Why should we use IOptions<T> when we can inject the configuration object directly?
There is no information on this page about why the `IOptions<T>` interface exists, what purpose it serves, or why we should need to inject an `IOptions<T>` instance instead of a `T` instance for accessing configuration.

For comparison, a similar example in Java Spring Boot would look like this:
```java
@ConfigurationProperties
public class PositionOptions
{
    public string title;
    public string name;
}

@Configuration
@EnableConfigurationProperties(PositionOptions.class)
public class MyAppConfiguration { }

public class Test2Model
{
    public Test2Model(PositionOptions options) { }
}
```

Note that the parameter to the model constructor is the type `PositionOptions`; there is no need for any wrapper type.

There just doesn't seem to be any value in wrapping the actual options type in `IOptions<>`. But I want to believe there is a rationale for this that I am missing, and I would love to see that rationale added to this page.

P.S. An issue was opened against the corresponding page for ASP.NET Core in November 2018 titled ["Why should we use IOptions<T> when we can inject the configuration object directly"](https://github.com/dotnet/AspNetCore.Docs/issues/9663). There was some dicussion, and the issue was eventually closed after a docs update, ["Emphasize IOptionsMonitor in Options topic + updates"](https://github.com/dotnet/AspNetCore.Docs/pull/9684). It was a very large update, and there is no way for me to understand what part of that update was supposed to apply to that issue. But reading the page now, I don't see anything that addresses that issue's question.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 433988ea-fda9-aa83-f6ee-221a5521a422
* Version Independent ID: 00449cdc-b8fc-a9ff-3f20-0ef0994db354
* Content: [Options pattern in .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/options)
* Content Source: [docs/core/extensions/options.md](https://github.com/dotnet/docs/blob/master/docs/core/extensions/options.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Added rationale section per customer feedback, and in correspondence to @davidfowl outlined thoughts.

Fixes #22886

### CODE REVIEW

File: docs/core/extensions/options.md [hunk 1]
```diff
@@ -101,11 +101,21 @@ In the preceding code, changes to the JSON configuration file after the app has
...
 
 <xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601> is used by <xref:Microsoft.Extensions.Options.IOptionsMonitor%601> to cache `TOptions` instances. The <xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601> invalidates options instances in the monitor so that the value is recomputed (<xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601.TryRemove%2A>). Values can be manually introduced with <xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601.TryAdd%2A>. The <xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601.Clear%2A> method is used when all named instances should be recreated on demand.
 
+### Options interfaces rationale
+
+There is a common question among developers regarding options interfaces:
+
+> Why should we use `IOptions<T>` when we can inject the configuration object directly?
```
Comment (BillWagner) [inline docs/core/extensions/options.md:113]: I would re-state this more declaratively.  Here's a first rough draft. Take what makes sense, and reword as you see fit:

```suggestion
### Options interfaces benefits

Using a generic wrapper type gives us the ability to decouple the lifetime of the option from the DI container. The <xref:Microsoft.Extensions.Options.IOptions%601.Value?displayProperty=nameWithType> interface provides a layer of abstraction, including generic constraints, on your options type. This provides the following benefits:
```

**Review by BillWagner — APPROVED** [review]
This LGTM @IEvangelist 

I had a wording suggestion (season to taste). Then, once the build is working, you can :shipit:

File: docs/core/extensions/options.md [hunk 2]
```diff
@@ -101,11 +101,19 @@ In the preceding code, changes to the JSON configuration file after the app has
...
 
 <xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601> is used by <xref:Microsoft.Extensions.Options.IOptionsMonitor%601> to cache `TOptions` instances. The <xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601> invalidates options instances in the monitor so that the value is recomputed (<xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601.TryRemove%2A>). Values can be manually introduced with <xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601.TryAdd%2A>. The <xref:Microsoft.Extensions.Options.IOptionsMonitorCache%601.Clear%2A> method is used when all named instances should be recreated on demand.
 
+### Options interfaces benefits
+
+Using a generic wrapper type gives us the ability to decouple the lifetime of the option from the DI container. The <xref:Microsoft.Extensions.Options.IOptions%601.Value?displayProperty=nameWithType> interface provides a layer of abstraction, including generic constraints, on your options type. This provides the following benefits:
```
Comment (gewarren) [inline docs/core/extensions/options.md:111]:
```suggestion
Using a generic wrapper type gives you the ability to decouple the lifetime of the option from the DI container. The <xref:Microsoft.Extensions.Options.IOptions%601.Value?displayProperty=nameWithType> interface provides a layer of abstraction, including generic constraints, on your options type. This provides the following benefits:
```

File: docs/core/extensions/options.md [hunk 3]
```diff
@@ -101,11 +101,19 @@ In the preceding code, changes to the JSON configuration file after the app has
...
+
+- The evaluation of the `T` configuration instance is deferred to the accessing of <xref:Microsoft.Extensions.Options.IOptions%601.Value?displayProperty=nameWithType>, rather than when it is injected. This is important because you can consume the `T` option from various places and choose the lifetime semantics without changing anything about `T`.
+- When registering options of type `T`, you do not need to explicitly register the `T` type. This is a convenience when you're [authoring a library](options-library-authors.md) with simple defaults where you don't want to force the caller to register options into the DI container with a specific lifetime.
```
Comment (gewarren) [inline docs/core/extensions/options.md:114]:
```suggestion
- When registering options of type `T`, you do not need to explicitly register the `T` type. This is a convenience when you're [authoring a library](options-library-authors.md) with simple defaults, and you don't want to force the caller to register options into the DI container with a specific lifetime.
```

File: docs/core/extensions/options.md [hunk 4]
```diff
@@ -101,11 +101,19 @@ In the preceding code, changes to the JSON configuration file after the app has
...
 
+### Options interfaces benefits
+
+Using a generic wrapper type gives you the ability to decouple the lifetime of the option from the DI container. The <xref:Microsoft.Extensions.Options.IOptions%601.Value?displayProperty=nameWithType> interface provides a layer of abstraction, including generic constraints, on your options type. This provides the following benefits:
+
+- The evaluation of the `T` configuration instance is deferred to the accessing of <xref:Microsoft.Extensions.Options.IOptions%601.Value?displayProperty=nameWithType>, rather than when it is injected. This is important because you can consume the `T` option from various places and choose the lifetime semantics without changing anything about `T`.
+- When registering options of type `T`, you do not need to explicitly register the `T` type. This is a convenience when you're [authoring a library](options-library-authors.md) with simple defaults, and you don't want to force the caller to register options into the DI container with a specific lifetime.
```
Comment (guardrex) [inline docs/core/extensions/options.md:114]: Hi @IEvangelist ... Just passing thru and looking at a few things over here. I noticed a little comma splice here in the list of two objects for "when" in the second sentence ...

```suggestion
- When registering options of type `T`, you do not need to explicitly register the `T` type. This is a convenience when you're [authoring a library](options-library-authors.md) with simple defaults and you don't want to force the caller to register options into the DI container with a specific lifetime.
```

**Pull request merged**